src/libraries/System.Collections/tests/Generic/PriorityQueue/PriorityQueue.Generic.Tests.cs

```diff
@@ -0,0 +1,307 @@
+                Assert.Equal(expectedPeekPriority, actualTryPeekPriority);
+                Assert.Equal(expectedPeekPriority, actualTryDequeuePriority);
+            }
+
+            Assert.Equal(expected: 0, queue.Count);
+            Assert.False(queue.TryPeek(out _, out _));
+            Assert.False(queue.TryDequeue(out _, out _));
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void PriorityQueue_Generic_EnqueueRange_IEnumerable(int count)
+        {
+            HashSet<(TElement, TPriority)> itemsToEnqueue = this.GenericIEnumerableFactory(count).ToHashSet();
+            PriorityQueue<TElement, TPriority> queue = new PriorityQueue<TElement, TPriority>();
+
+            queue.EnqueueRange(itemsToEnqueue);
+
+            Assert.True(itemsToEnqueue.SetEquals(queue.UnorderedItems));
+        }
+
+        #endregion
+
+        #region Clear
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void PriorityQueue_Generic_Clear(int count)
+        {
+            PriorityQueue<TElement, TPriority> queue = this.GenericPriorityQueueFactory(initialCapacity: 0, count, out _);
+
+            Assert.Equal(count, queue.Count);
+            queue.Clear();
+            Assert.Equal(expected: 0, queue.Count);
+        }
+
+        #endregion
+
+        #region EnsureCapacity, TrimExcess
+
+        [Theory]
+        [InlineData(0, 0)]
+        [InlineData(0, 5)]
+        [InlineData(1, 1)]
+        [InlineData(3, 100)]
+        public void PriorityQueue_Generic_TrimExcess_ValidQueueThatHasntBeenRemovedFrom(int initialCapacity, int count)
+        {
+            PriorityQueue<TElement, TPriority> queue = this.GenericPriorityQueueFactory(initialCapacity, count, out _);
+            queue.TrimExcess();
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void PriorityQueue_Generic_TrimExcess_Repeatedly(int count)
+        {
+            PriorityQueue<TElement, TPriority> queue = this.GenericPriorityQueueFactory(initialCapacity: 0, count, out _);
+
+            Assert.Equal(count, queue.Count);
+            queue.TrimExcess();
+            queue.TrimExcess();
+            queue.TrimExcess();
+            Assert.Equal(count, queue.Count);
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidPositiveCollectionSizes))]
+        public void PriorityQueue_Generic_EnsureCapacityAndTrimExcess(int count)
+        {
+            IReadOnlyCollection<(TElement, TPriority)> itemsToEnqueue = this.GenericIEnumerableFactory(count).ToArray();
+            PriorityQueue<TElement, TPriority> queue = new PriorityQueue<TElement, TPriority>();
+            int expectedCount = 0;
+            Random random = new Random(Seed: 34);
+            int getNextEnsureCapacity() => random.Next(0, count * 2);
+            void trimAndEnsureCapacity()
+            {
+                queue.TrimExcess();
```

Comment:
We might be able to make stronger assertions about `EnsureCapacity` and `TrimExcess` using reflection to read the actual buffer capacity of the queue.